src/main/java/Deadline.java

```diff
@@ -0,0 +1,20 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
+    protected LocalDate by;
+    static DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM d yyyy");
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(format) + ")";
```

Comment:
As [D] shows up in multiple places, you may want to consider making it a constant within the class